cg/meta/clean/clean_flow_cells.py

```diff
@@ -0,0 +1,178 @@
+        return is_directory_older_than_days_old(
+            directory_path=self.flow_cell.path,
+            days_old=TWENTY_ONE_DAYS,
+            current_time=self.current_time,
+        )
+
+    def is_flow_cell_in_statusdb(self) -> bool:
+        """Check if flow cell is in statusdb."""
+        return bool(self.get_flow_cell_from_status_db())
+
+    def is_flow_cell_backed_up(self) -> bool:
+        """Check if flow cell is backed up on PDC."""
+        return bool(self.get_flow_cell_from_status_db().has_backup)
+
+    def has_sequencing_metrics_in_statusdb(self) -> bool:
+        """Check if a flow cell has entries in the SampleLaneSequencingMetrics table."""
+        return bool(self.get_sequencing_metrics_for_flow_cell())
+
+    def has_sample_sheet_in_housekeeper(self) -> bool:
+        """Check if the flow cell has a sample sheet in housekeeper."""
+        return bool(self.flow_cell.get_sample_sheet_path_hk())
+
+    def has_fastq_files_for_samples_in_housekeeper(self) -> bool:
+        """Check if all samples on the flow cell have fastq files in housekeeper."""
+        return bool(self.has_files_for_samples_on_flow_cell_with_tag(tag=SequencingFileTag.FASTQ))
+
+    def has_spring_files_for_samples_in_housekeeper(self) -> bool:
+        """Check if all samples on the flow cell have SPRING files in housekeeper."""
+        return bool(self.has_files_for_samples_on_flow_cell_with_tag(tag=SequencingFileTag.SPRING))
+
+    def has_spring_meta_data_files_for_samples_in_housekeeper(self) -> bool:
+        """Check if all samples on the flow cell have SPRING metadata files in housekeeper."""
+        return bool(
+            self.has_files_for_samples_on_flow_cell_with_tag(tag=SequencingFileTag.SPRING_METADATA)
+        )
+
+    def get_flow_cell_from_status_db(self) -> Optional[Flowcell]:
+        """
+        Get the flow cell entry from StatusDB.
+        Raises:
+            ValueError if the flow cell is not found in StatusDB.
+        """
+        flow_cell: Flowcell = self.status_db.get_flow_cell_by_name(self.flow_cell.id)
+        if not flow_cell:
+            raise ValueError(f"Flow cell {self.flow_cell.id} not found in StatusDB.")
+        return flow_cell
+
+    def get_sequencing_metrics_for_flow_cell(self) -> Optional[List[SampleLaneSequencingMetrics]]:
+        """
+        Get the SampleLaneSequencingMetrics entries for a flow cell.
+        Raises:
+              Value error if no SampleLaneSequencingMetrics are found in StatusDB.
+        """
+        metrics: List[
+            SampleLaneSequencingMetrics
+        ] = self.status_db.get_sample_lane_sequencing_metrics_by_flow_cell_name(self.flow_cell.id)
+        if not metrics:
+            raise ValueError(
+                f"No SampleLaneSequencingMetrics found for {self.flow_cell.id} in StatusDB."
+            )
+        return metrics
+
+    def has_files_for_samples_on_flow_cell_with_tag(self, tag: str) -> Optional[List[File]]:
```

Comment:
This function should return a bool and not raise any errors, as it is only used as input for checker functions